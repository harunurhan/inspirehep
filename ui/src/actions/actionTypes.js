export const SEARCH_REQUEST = 'SEARCH_REQUEST';
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS';
export const SEARCH_ERROR = 'SEARCH_ERROR';

export const LITERATURE_REQUEST = 'LITERATURE_REQUEST';
export const LITERATURE_SUCCESS = 'LITERATURE_SUCCESS';
export const LITERATURE_ERROR = 'LITERATURE_ERROR';
