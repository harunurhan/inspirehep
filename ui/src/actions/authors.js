import { stringify } from 'qs';

import {
  AUTHOR_REQUEST,
  AUTHOR_SUCCESS,
  AUTHOR_ERROR,
  AUTHOR_PUBLICATIONS_REQUEST,
  AUTHOR_PUBLICATIONS_SUCCESS,
  AUTHOR_PUBLICATIONS_ERROR,
  AUTHOR_PUBLICATIONS_FACETS_REQUEST,
  AUTHOR_PUBLICATIONS_FACETS_SUCCESS,
  AUTHOR_PUBLICATIONS_FACETS_ERROR,
} from './actionTypes';
import { UI_SERIALIZER_REQUEST_OPTIONS } from '../common/http';

function fetchingAuthor(recordId) {
  return {
    type: AUTHOR_REQUEST,
    payload: { recordId },
  };
}

function fetchAuthorSuccess(result) {
  return {
    type: AUTHOR_SUCCESS,
    payload: result,
  };
}

function fetchAuthorError(error) {
  return {
    type: AUTHOR_ERROR,
    payload: error,
  };
}

export default function fetchAuthor(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingAuthor(recordId));
    try {
      const response = await http.get(
        `/authors/${recordId}`,
        UI_SERIALIZER_REQUEST_OPTIONS
      );
      dispatch(fetchAuthorSuccess(response.data));
    } catch (error) {
      dispatch(fetchAuthorError(error.response && error.response.data));
    }
  };
}

function getAuthorPublicationsQuery(authorsState, newQuery) {
  const query = {
    ...authorsState.getIn(['publications', 'query']).toJS(),
    ...newQuery,
  };
  return query;
}

function fetchingAuthorPublications(query) {
  return {
    type: AUTHOR_PUBLICATIONS_REQUEST,
    payload: query,
  };
}

function fetchAuthorPublicationSuccess(result) {
  return {
    type: AUTHOR_PUBLICATIONS_SUCCESS,
    payload: result,
  };
}

function fetchAuthorPublicationsError(error) {
  return {
    type: AUTHOR_PUBLICATIONS_ERROR,
    payload: error,
  };
}

export function fetchAuthorPulications(newQuery = {}) {
  return async (dispatch, getState, http) => {
    const { authors } = getState();

    const query = getAuthorPublicationsQuery(authors, newQuery);
    dispatch(fetchingAuthorPublications(query));

    const authorFacetValue = authors.getIn([
      'data',
      'metadata',
      'facet_author_name',
    ]);
    const queryString = stringify(query, { indices: false });
    try {
      const response = await http.get(
        `/literature?author=${authorFacetValue}&${queryString}`,
        UI_SERIALIZER_REQUEST_OPTIONS
      );
      dispatch(fetchAuthorPublicationSuccess(response.data));
    } catch (error) {
      dispatch(
        fetchAuthorPublicationsError(error.response && error.response.data)
      );
    }
  };
}

function fetchingAuthorPublicationsFacets(query) {
  return {
    type: AUTHOR_PUBLICATIONS_FACETS_REQUEST,
    payload: query,
  };
}

function fetchAuthorPublicationFacetsSuccess(result) {
  return {
    type: AUTHOR_PUBLICATIONS_FACETS_SUCCESS,
    payload: result,
  };
}

function fetchAuthorPublicationsFacetsError(error) {
  return {
    type: AUTHOR_PUBLICATIONS_FACETS_ERROR,
    payload: error,
  };
}

export function fetchAuthorPulicationsFacets(newQuery = {}) {
  return async (dispatch, getState, http) => {
    const { authors } = getState();

    const query = getAuthorPublicationsQuery(authors, newQuery);

    dispatch(fetchingAuthorPublicationsFacets(query));

    const authorFacetValue = authors.getIn([
      'data',
      'metadata',
      'facet_author_name',
    ]);
    const queryString = stringify(query, { indices: false });
    try {
      const response = await http.get(
        `/literature/facets?facet_name=hep-author-publication&exclude_author_value=${authorFacetValue}&author=${authorFacetValue}&${queryString}`
      );
      dispatch(fetchAuthorPublicationFacetsSuccess(response.data));
    } catch (error) {
      dispatch(
        fetchAuthorPublicationsFacetsError(
          error.response && error.response.data
        )
      );
    }
  };
}
