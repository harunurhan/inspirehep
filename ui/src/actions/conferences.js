import {
  CONFERENCE_REQUEST,
  CONFERENCE_SUCCESS,
  CONFERENCE_ERROR,
} from './actionTypes';
import { UI_SERIALIZER_REQUEST_OPTIONS } from '../common/http';
import generateRecordFetchAction from './recordsFactory';
import { CONFERENCES_PID_TYPE } from '../common/constants';

const fetchConference = generateRecordFetchAction({
  pidType: CONFERENCES_PID_TYPE,
  fetchingActionActionType: CONFERENCE_REQUEST,
  fecthSuccessActionType: CONFERENCE_SUCCESS,
  fetchErrorActionType: CONFERENCE_ERROR,
  requestOptions: UI_SERIALIZER_REQUEST_OPTIONS,
});

export default fetchConference;
