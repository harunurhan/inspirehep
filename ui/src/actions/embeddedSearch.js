import { stringify } from 'qs';

import {
  EMBEDDED_SEARCH_REQUEST,
  EMBEDDED_SEARCH_SUCCESS,
  EMBEDDED_SEARCH_ERROR,
  EMBEDDED_SEARCH_AGGREGATIONS_REQUEST,
  EMBEDDED_SEARCH_AGGREGATIONS_SUCCESS,
  EMBEDDED_SEARCH_AGGREGATIONS_ERROR,
  UPDATE_EMBEDDED_SEARCH_QUERY,
  EMBEDDED_SEARCH_SET_OPTIONS,
} from './actionTypes';
import { UI_SERIALIZER_REQUEST_OPTIONS } from '../common/http';
import { httpErrorToActionPayload } from '../common/utils';

function searching() {
  return {
    type: EMBEDDED_SEARCH_REQUEST,
  };
}

function searchSuccess(result) {
  return {
    type: EMBEDDED_SEARCH_SUCCESS,
    payload: result,
  };
}

function searchError(error) {
  return {
    type: EMBEDDED_SEARCH_ERROR,
    payload: error,
  };
}

function fetchingSearchAggregations() {
  return {
    type: EMBEDDED_SEARCH_AGGREGATIONS_REQUEST,
  };
}

function searchAggregationsSuccess(result) {
  return {
    type: EMBEDDED_SEARCH_AGGREGATIONS_SUCCESS,
    payload: result,
  };
}

function searchAggregationsError(error) {
  return {
    type: EMBEDDED_SEARCH_AGGREGATIONS_ERROR,
    payload: error,
  };
}

function getSearchQueryFromState(state) {
  const { embeddedSearch } = state;
  const query = embeddedSearch.get('query').toJS();
  return query;
}

function getAggregationsQueryFromState(state) {
  const { embeddedSearch } = state;
  const baseAggregationsQuery = embeddedSearch
    .get('baseAggregationsQuery')
    .toJS();
  return {
    ...baseAggregationsQuery,
    ...getSearchQueryFromState(state),
  };
}

async function fetchSearchResults(dispatch, state, http) {
  const pidType = state.embeddedSearch.get('pidType');
  const query = getSearchQueryFromState(state);
  const queryString = stringify(query, { indices: false });

  const url = `/${pidType}?${queryString}`;
  dispatch(searching());
  try {
    const response = await http.get(url, UI_SERIALIZER_REQUEST_OPTIONS);
    dispatch(searchSuccess(response.data));
  } catch (error) {
    const payload = httpErrorToActionPayload(error);
    dispatch(searchError(payload));
  }
}

async function fetchSearchAggregations(dispatch, state, http) {
  const pidType = state.embeddedSearch.get('pidType');
  const query = getAggregationsQueryFromState(state);

  const queryString = stringify(query, { indices: false });
  const url = `/${pidType}/facets?${queryString}`;

  dispatch(fetchingSearchAggregations());
  try {
    const response = await http.get(url);
    dispatch(searchAggregationsSuccess(response.data));
  } catch (error) {
    const payload = httpErrorToActionPayload(error);
    dispatch(searchAggregationsError(payload));
  }
}

function updateSearchQuery(query) {
  return {
    type: UPDATE_EMBEDDED_SEARCH_QUERY,
    payload: { query },
  };
}

export function search(newQuery) {
  return async (dispatch, getState, http) => {
    await dispatch(updateSearchQuery(newQuery));
    // TODO: conditional calls like in `searchDispatcher`
    const state = getState();
    fetchSearchAggregations(dispatch, state, http);
    fetchSearchResults(dispatch, state, http);
  };
}

export function setOptions(options) {
  return {
    type: EMBEDDED_SEARCH_SET_OPTIONS,
    payload: options,
  };
}
