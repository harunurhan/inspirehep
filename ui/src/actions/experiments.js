import {
  EXPERIMENT_REQUEST,
  EXPERIMENT_SUCCESS,
  EXPERIMENT_ERROR,
} from './actionTypes';
import { UI_SERIALIZER_REQUEST_OPTIONS } from '../common/http';
import { EXPERIMENTS_PID_TYPE } from '../common/constants';
import generateRecordFetchAction from './recordsFactory';

const fetchExperiment = generateRecordFetchAction({
  pidType: EXPERIMENTS_PID_TYPE,
  fetchingActionActionType: EXPERIMENT_REQUEST,
  fecthSuccessActionType: EXPERIMENT_SUCCESS,
  fetchErrorActionType: EXPERIMENT_ERROR,
  requestOptions: UI_SERIALIZER_REQUEST_OPTIONS,
});

export default fetchExperiment;
