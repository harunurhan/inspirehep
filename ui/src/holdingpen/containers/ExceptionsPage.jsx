import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import fetchExceptions from '../../actions/exceptions';
import toJS from '../../common/immutableToJS';
import ExceptionsDashboard from '../components/ExceptionsDashboard';

class ExceptionsPage extends Component {
  componentWillMount() {
    this.props.dispatch(fetchExceptions());
  }

  render() {
    return <ExceptionsDashboard exceptions={this.props.exceptions} />;
  }
}

ExceptionsPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  exceptions: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = state => ({
  exceptions: state.exceptions.get('data'),
});
const dispatchToProps = dispatch => ({ dispatch });

export default connect(mapStateToProps, dispatchToProps)(toJS(ExceptionsPage));
