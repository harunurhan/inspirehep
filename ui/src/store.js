import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
/* eslint-disable import/no-extraneous-dependencies */
import { composeWithDevTools } from 'redux-devtools-extension';

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import keepPreviousUrlMiddleware from './middlewares/keepPreviousUrl';
import {
  reHydrateRootStateFromStorage,
  createPersistToStorageMiddleware,
} from './middlewares/statePersister';
import searchDispatcherMiddleware from './middlewares/searchDispatcher';
import redirectToErrorPageMiddleware from './middlewares/redirectToErrorPage';
import actionTrackerMiddleware from './middlewares/actionTracker';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const reducersToPersist = ['ui', 'user'];

const PROD_MIDDLEWARES = [
  reduxRouterMiddleware,
  queryParamsParserMiddleware,
  keepPreviousUrlMiddleware,
  createPersistToStorageMiddleware(reducersToPersist),
  searchDispatcherMiddleware,
  redirectToErrorPageMiddleware,
  actionTrackerMiddleware,
  thunkMiddleware,
];

const DEV_MIDDLEWARES = [...PROD_MIDDLEWARES, createLogger()];

const withMiddlewares = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(...PROD_MIDDLEWARES);
  }
  return applyMiddleware(...DEV_MIDDLEWARES);
};

export const store = createStore(
  reducers,
  reHydrateRootStateFromStorage(reducersToPersist),
  composeWithDevTools(withMiddlewares())
);
