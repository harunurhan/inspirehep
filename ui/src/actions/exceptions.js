import { EXCEPTIONS_REQUEST, EXCEPTIONS_SUCCESS } from './actionTypes';
import data from '../holdingpen/report-errors.json';

function fetching() {
  return {
    type: EXCEPTIONS_REQUEST,
  };
}

function fetchSuccess(result) {
  return {
    type: EXCEPTIONS_SUCCESS,
    payload: result,
  };
}

export default function fetch() {
  return async dispatch => {
    dispatch(fetching());
    dispatch(fetchSuccess(data));
  };
}
