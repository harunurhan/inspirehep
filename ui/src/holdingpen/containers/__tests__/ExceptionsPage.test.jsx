import React from 'react';
import { shallow, mount } from 'enzyme';
import { fromJS } from 'immutable';
import { getStoreWithState } from '../../../fixtures/store';
import { EXCEPTIONS_REQUEST } from '../../../actions/actionTypes';
import ExceptionsPage from '../ExceptionsPage';

describe('ExceptionsPage', () => {
  it('renders initial state', () => {
    const store = getStoreWithState({
      exceptions: fromJS({
        data: [
          {
            collection: 'hep',
            error: 'Some error',
            recid: 123456,
          },
        ],
        loading: false,
      }),
    });

    const wrapper = shallow(<ExceptionsPage store={store} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders initial state when loading is true', () => {
    const store = getStoreWithState({
      exceptions: fromJS({
        data: [],
        loading: true,
      }),
    });

    const wrapper = shallow(<ExceptionsPage store={store} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches exceptions', () => {
    const store = getStoreWithState({
      exceptions: fromJS({
        data: [
          {
            collection: 'hep',
            error: 'Some error',
            recid: 123456,
          },
        ],
        loading: false,
      }),
    });
    mount(<ExceptionsPage store={store} />);
    const actions = store.getActions();
    expect(actions.some(action => action.type === EXCEPTIONS_REQUEST)).toBe(
      true
    );
  });
});
