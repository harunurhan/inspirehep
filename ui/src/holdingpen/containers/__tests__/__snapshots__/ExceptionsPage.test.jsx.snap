// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExceptionsPage renders initial state 1`] = `
<ExceptionsDashboard
  exceptions={
    Array [
      Object {
        "collection": "hep",
        "error": "Some error",
        "recid": 123456,
      },
    ]
  }
/>
`;
